wetmap/src/components/newModals/tripCreatorListModal/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { itineraries } from '../../../supabaseCalls/itinerarySupabaseCalls';
+import { updateDiveShop } from '../../../supabaseCalls/shopsSupabaseCalls';
+import { SelectedShopContext } from '../../contexts/selectedShopContext';
+import { UserProfileContext } from '../../contexts/userProfileContext';
+import { ModalContext } from '../../reusables/modal/context';
+import { ItineraryItem } from './types';
+import TripCreatorListView from './view';
+import { ModalHandleProps } from '../../reusables/modal/types';
+import DiveShopModal from '../shopModal/index';
+
+type TripCreatorListProps = Partial<ModalHandleProps>;
+
+export default function TripCreatorListModal(props: TripCreatorListProps) {
+  const { selectedShop } = useContext(SelectedShopContext);
+  const { profile } = useContext(UserProfileContext);
+  const modalContext = useContext(ModalContext);
+  const { modalShow } = useContext(ModalContext);
+
+  const [itineraryList, setItineraryList] = useState<ItineraryItem[]>([]);
+  const [selectedID, setSelectedID] = useState<number>(0);
+
+  useEffect(() => {
+    if (selectedShop) {
+      getItineraries(selectedShop.id);
+    }
+  }, [selectedShop]);
+
+  const getItineraries = async (IdNum: number) => {
+    try {
+      const itins = await itineraries(IdNum);
+      if (itins && itins.length > 0) {
+        setItineraryList(itins);
+      }
+    } catch (e) {
+      console.log({ title: 'Error', message: (e as Error).message });
+    }
+  };
+
+  const handleBackButton = async () => {
+    modalContext.modalCancel();
+    modalShow(DiveShopModal, {
+      size: 'large',
+    });
+  };
```

Comment:
Dont need to do this if you use `keepPreviousModal` when open TripCreatorListModal

Reply:
fixed